set.js

```diff
@@ -0,0 +1,90 @@
+class Set {
+  constructor(array) {
+    this.values = array || [];
+  }
+  add(value) {
+    this.values.push(value);
+  }
+  remove(value) {
+    this.values.splice(this.values.indexOf(value), 1);
+  }
+  has(value) {
+    return this.values.includes(value)
+  }
+  intersection(set) {
+    let result = new Set();
+    this.values.forEach(value => {
+      if(set.has(value)) {
+        result.add(value);
+      }
+    });
+    return result;
+  }
+  union(set) {
+    let result = new Set();
+    this.values.forEach(value => {
+      if(!result.has(value)) {
+        result.add(value);
+      }
+    });
+    set.values.forEach(value => {
+      if(!result.has(value)) {
+        result.add(value);
+      }
+    });
+    return result;
+  }
+  difference(set) {
+    let result = new Set();
+    this.values.forEach(value => {
+      if(!set.has(value)) {
+        result.add(value);
+      }
+    });
+    set.values.forEach(value => {
+      if(!this.has(value)) {
+        result.add(value);
+      }
+    });
+    return result;
+  }
+  static staticIntersection(set1, set2) {
```

Comment:
you can call the static method `intersection` also.

You shouldn't have to duplicate this code